pkg/api/login.go

```diff
@@ -240,19 +240,29 @@ func (hs *HTTPServer) loginUserWithUser(user *user.User, c *contextmodel.ReqCont
 }
 
 func (hs *HTTPServer) Logout(c *contextmodel.ReqContext) {
-	// If SAML is enabled and this is a SAML user use saml logout
-	if hs.samlSingleLogoutEnabled() {
-		getAuthQuery := loginservice.GetAuthInfoQuery{UserId: c.UserID}
-		if authInfo, err := hs.authInfoService.GetAuthInfo(c.Req.Context(), &getAuthQuery); err == nil {
+	oauthProviderSignoutRedirectUrl := ""
+	getAuthQuery := loginservice.GetAuthInfoQuery{UserId: c.UserID}
+	authInfo, err := hs.authInfoService.GetAuthInfo(c.Req.Context(), &getAuthQuery)
+	if err == nil {
+		if hs.samlSingleLogoutEnabled() {
 			if authInfo.AuthModule == loginservice.SAMLAuthModule {
 				c.Redirect(hs.Cfg.AppSubURL + "/logout/saml")
 				return
 			}
 		}
+		oauthProvider := hs.SocialService.GetOAuthInfoProvider(strings.TrimPrefix(authInfo.AuthModule, "oauth_"))
+		oauthProviderSignoutRedirectUrl = oauthProvider.SignoutRedirectUrl
 	}
 
+	signOutRedirectUrl := hs.Cfg.SignoutRedirectUrl
```

Comment:
I'm not so sure the init of signOutRedirectUrl from this line is really needed. I think it's enough to init that var at line #261 and slightly modify the logging below to use hs.Cfg.SignoutRedirectUrl.

Reply:
Agree, will make changes